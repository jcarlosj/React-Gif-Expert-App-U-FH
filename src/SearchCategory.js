import React, { useState } from 'react'

export const SearchCategory = () => {

    const [ inputValue, setInputValue] = useState( '' );

    const handleInputChange = event => {
        console.log( event.target.value );
        setInputValue( event.target.value );
    }

    return (
        <form>
            <pre>{ JSON.stringify( inputValue, null, 4 ) }</pre>
            <input 
                type="text" 
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}
