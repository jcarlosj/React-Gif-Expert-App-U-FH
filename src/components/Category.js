import React from 'react';

export const Category = ({ category }) => {

    const requestToGiphy = async () => {    //  https://developers.giphy.com/
        const 
            url = `https://api.giphy.com/v1/gifs/search?q=superman&limit=9&api_key=6t2zjzJn3x3LILIFUpF6GKKcR95JTJhK`,
            response = await fetch( url ),
            { data } = await response.json();
        // console.log( data );

        const gifs = data.map( image => {
            return {
                id: image.id,
                title: image.title,
                url: image.images?.downsized_medium.url
            };
        });
        console.log( gifs );
    }
    requestToGiphy();

    return (
        <>
            <h3>{ category }</h3>
        </>
    )
}
