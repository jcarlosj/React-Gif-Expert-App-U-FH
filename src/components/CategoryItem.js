import React from 'react';

export const CategoryItem = ( { title, url } ) => {
    return (
        <div className="card animate__animated animate__bounceIn">
            <img src={ url } alt={ title } title={ title } />        
            <h4>{ title }</h4>
        </div>
   )
}