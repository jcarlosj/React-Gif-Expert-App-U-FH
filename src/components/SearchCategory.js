import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const SearchCategory = ({ setCategories }) => {

    const [ inputValue, setInputValue] = useState( '' );

    const handleInputChange = event => {
        // console.log( event.target.value );
        setInputValue( event.target.value );
    }

    const handleSubmit = event => {
        event.preventDefault();

        if( inputValue.trim().length > 2 ) {
            setCategories( categories => [ ...categories, inputValue ] );
            setInputValue( '' );
        }
        
    }

    return (
        <form
            onSubmit={ handleSubmit }
        >
            {/* <pre>{ JSON.stringify( inputValue, null, 4 ) }</pre> */}
            <input 
                type="text" 
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

SearchCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
