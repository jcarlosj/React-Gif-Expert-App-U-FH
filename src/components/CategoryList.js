import React from 'react';

/** Hooks */
import { useFetchGifs } from '../hooks/useFecthGifs';

/** Component */
import { CategoryItem } from './CategoryItem';

/** Functional Components */
export const CategoryList = ({ category }) => {

    const { loading, data:images } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>

            {   loading && <p className="animate__animated animate__flash">Cargando...</p> }

            <div className="card-grid">
            
                {   images.map( image => (
                        <CategoryItem 
                            key={ image.id }
                            { ...image }    //  Spread
                        />
                    ))
                }
                
            </div>
        </>
    )
}
