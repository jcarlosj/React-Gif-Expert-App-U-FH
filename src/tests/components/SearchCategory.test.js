import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { SearchCategory } from '../../components/SearchCategory';

describe( '<SearchCategory />', () => {

    const
        setCategories = () => {}, 
        wrapper = shallow(
            <SearchCategory setCategories={ setCategories } />
        );

    test( 'debe desplegar correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

    test( 'debe cambiar contenido de input', () => {
        
        const
            value = 'Testing input',
            $input = wrapper.find( 'input' );

        $input.simulate( 'change', { target: { value } } );
        expect( wrapper.find( 'pre' ).text().trim() ).toBe( JSON.stringify( value, null, 4 ) );

    });
    
});