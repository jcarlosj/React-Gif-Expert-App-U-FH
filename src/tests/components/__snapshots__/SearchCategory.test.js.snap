// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchCategory /> debe desplegar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <pre>
    ""
  </pre>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
